### PR TITLE
make docker-compose file

### LINKED ISSUE #3: Setup development environment by Docker
## Summary

We have to setup development environment by Docker.
So, we need to prepare the following:

- `Dockerfile`: for User auth API
- `docker-compose.yml `: for combine API App, RDB, etc...
  - RDB: `MySQL`
  - API App: `fastapi`

### LINKED ISSUE #6: Setup development environment
## Summary

We have to setup development environment for develop apps.

## Requirements 

- `Python 3.8`
- `FastAPI` (https://pypi.org/project/fastapi/)
- `jwt` (https://pypi.org/project/jwt/)

### PR DESCRIPTION
**Pull request by oreotabetai**
Install python3.8 and pipenv.
Install fastapi and jwt by pipenv.

Start this container to command "docker-compose up -d" in this directry.

resolved #3
resolved #6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.8
```
Comment (yamakenji24) [inline Dockerfile]: Maybe using just python image could be heavy, so why don't we use alpine instead?

**Review by yamakenji24 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (wakatakeru) [reply Dockerfile]: i recommend `python` image with `3.8-alpine` tag.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+WORKDIR /app
+
+COPY . .
+
+RUN  pip install pipenv
+
+RUN pipenv install fastapi jwt
```
Comment (wakatakeru) [inline Dockerfile]: Libralies shold be install by `pip` in Docker container, without using pipenv.
I recommend to use `requirements.txt` for definition of dependencies.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+version: '3'
+services:
+  uajp:
+    build: .
+    tty: true
+    depends_on:
+      - mysql_db
+  
+  mysql_db:
```
Comment (wakatakeru) [inline docker-compose.yml]: `mysql_db` should be `mysql`.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    image: mysql
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+    #volumes:
```
Comment (wakatakeru) [inline docker-compose.yml]: i recommend setting Docker volume for data persistence.

https://docs.docker.com/compose/compose-file/#volumes

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    depends_on:
+      - mysql_db
+  
+  mysql_db:
+    image: mysql
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+    #volumes:  
+    ports:
+    - "3306:3306"
```
Comment (wakatakeru) [inline docker-compose.yml]: I don't think it's necessary to expose port 3306 to `localhost`.

File: Dockerfile (hunk 2 above)
Comment (wakatakeru) [reply Dockerfile]: i recommend adding newline.

**Review by wakatakeru — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (wakatakeru) [reply Dockerfile]:
```
RUN pip install -r requirements.txt
```

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+version: '3'
+services:
+  uajp:
+    build: .
+    tty: true
+    depends_on:
+      - mysql_db
+
```
Comment (wakatakeru) [inline docker-compose.yml]: Our application uses port of `3000`. So, you should expose `3000` to `localhost`.

**Comment by wakatakeru** [comment]
- resolved #3 
- resolved #6

**Comment by wakatakeru** [comment]
LGTM!

**Review by yamakenji24 — APPROVED** [review]
LGTM

**Review by wakatakeru — APPROVED** [review]

**Pull request merged**